sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/RequireFailForTryCatchInJunitCheck.java

```diff
@@ -382,6 +385,22 @@ private boolean isValidFail(DetailAST expression) {
         return result;
     }
 
+    /**
+     * Returns the last method in the method chain.
+     * For {@code Truth.assert_().withMessage().withMessage().fail()} this will be {@code fail}.
+     *
+     * @param methodCall the method call AST
+     * @return the AST of the last method call in the chain
+     */
+    private static DetailAST skipToTheLastMethodInChain(DetailAST methodCall) {
+        DetailAST result = methodCall;
+        while (result.getFirstChild().getType() == TokenTypes.DOT
+                && result.getFirstChild().getFirstChild().getType() == TokenTypes.METHOD_CALL) {
+            result = result.getFirstChild().getFirstChild();
```

Comment:
Just a thought. Does it make sense to split result into 2 variables since there seems to be a lot of `firstChild` calls?

init:
````
result = methodCall;
child = methodCall.getFirstChild()`
````
inside loop:
````
result = child.getFirstChild()
child = result.getFirstChild()
````

Reply:
done